lib/p2p/NodeList.ts

```diff
@@ -170,23 +170,26 @@ class NodeList extends EventEmitter {
     const node = this.nodes.get(nodePubKey);
 
     if (node) {
-      const promises: PromiseLike<any>[] = [
-        this.repository.addReputationEvent({ event, nodeId: node.id }),
-      ];
+      const promises: PromiseLike<any>[] = [];
 
       this.updateReputationScore(node, event);
 
-      if (node.reputationScore < NodeList.BAN_THRESHOLD) {
+      if (node.reputationScore < NodeList.BAN_THRESHOLD && !node.banned) {
         promises.push(this.setBanStatus(node, true));
 
-        const events = await this.repository.getReputationEvents(node);
-        this.emit('node.ban', nodePubKey, events);
-      } else if (node.banned) {
+        const negativeReputationEvents = (await this.repository.getReputationEvents(node))
+          .filter(e => reputationEventWeight[e.event] < 0).slice(0, 9)
```

Comment:
Just wondering why do we need to send up to last 10 reputation events to the peer rather than just the final one that triggered the ban?

Reply:
That's just the way it was set up originally, it might be helpful in the case of repeated events triggering the failure so the peer has a better idea of why it was banned. I am fine with only sending the most recent event, however.

Reply:
That makes sense. Was just curious. I'm OK with leaving it with up to 10 last negative events.